source/common/http/async_client_impl.cc

```diff
@@ -79,6 +79,9 @@ void AsyncStreamImpl::encodeHeaders(HeaderMapPtr&& headers, bool end_stream) {
       },
       nullptr);
 #endif
+  if (remote_closed_) {
```

Comment:
Sorry, this PR is stale, I think I had some dirty client state locally that had remove from `encode{Headers,Trailers}` the guard.